NotificationSystem.Tests/EmailSenderTests.cs

```diff
@@ -0,0 +1,232 @@
+
+        }
+
+        [Test]
+        public void ValidatePayload_EmptyTextContent_False()
+        {
+            // Arrange
+            EmailNotification notification = new EmailNotification("test@gmail.com", "");
+            string payload = Newtonsoft.Json.JsonConvert.SerializeObject(notification);
+            string emailFromString = "test@gmail.com";
+            EmailSender emailSender = new EmailSender(null, emailFromString);
+
+            // Act
+            emailSender.ParsePayload(payload);
+            bool result = emailSender.ValidatePayload();
+
+            // Assert
+            Assert.AreEqual(false, result);
+        }
+
+        [Test]
+        public void EmailSending_WithValidApiKey_ValidEmail()
+        {
+            // Arrange
+            EmailNotification notification = new EmailNotification("test@example.com", "plain text content");
+            string payload = Newtonsoft.Json.JsonConvert.SerializeObject(notification);
+
+            string jsonConfig = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "..\\..\\..\\..\\NotificationSystem\\appsettings.Development.json"));
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddJsonFile(jsonConfig, true, true)
+                .AddEnvironmentVariables()
+                .Build();
+
+            Config config = configuration.GetSection("App").Get<Config>();
+            SendGridClient sendGridClient = new SendGridClient(config.SendGrid.ApiKey);
+
+            // Act
+            EmailSender sender = new EmailSender(sendGridClient, config.SendGrid.EmailFrom, true);
+            sender.ParsePayload(payload);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.OK, sender.ExecuteTask().StatusCode);
+        }
+
+        [Test]
+        public void EmailSending_WithInvalidApiKey_ValidEmail()
+        {
+            // Arrange
+            EmailNotification notification = new EmailNotification("test@example.com", "plain text content");
+            string payload = Newtonsoft.Json.JsonConvert.SerializeObject(notification);
+
+            string jsonConfig = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "..\\..\\..\\..\\NotificationSystem\\appsettings.Development.json"));
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddJsonFile(jsonConfig, true, true)
+                .AddEnvironmentVariables()
+                .Build();
+
+            Config config = configuration.GetSection("App").Get<Config>();
+
+            SendGridClient sendGridClient = new SendGridClient("test");
+
+            // Act
+            EmailSender sender = new EmailSender(sendGridClient, config.SendGrid.EmailFrom, true);
+            sender.ParsePayload(payload);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.Unauthorized, sender.ExecuteTask().StatusCode);
+        }
+
+        [Test]
+        public void EmailSending_WithValidApiKey_InvalidEmail()
```

Comment:
Require sendgrid API key, may be unclear. I think it would be useful to add a comment here.